lib/email/receiver.rb

```diff
@@ -577,6 +577,7 @@ def process_destination(destination, user, body, elided, hidden_reason_id)
 
       case destination[:type]
       when :group
+        enable_email_pm_setting(user)
```

Comment:
`process_destination` won't be called when the `find_related_post_with_key` site setting is `false`.